pkg/storage/snapshot/restore.go

```diff
@@ -917,6 +938,27 @@ func (ctrl *VMRestoreController) getDV(namespace, name string) (*v1beta1.DataVol
 	return obj.(*v1beta1.DataVolume).DeepCopy(), nil
 }
 
+func (t *vmRestoreTarget) getDeletedDV(newTemplates []kubevirtv1.DataVolumeTemplateSpec) []string {
+	var deletedDataVolumes []string
+	for _, cdv := range t.vm.Spec.DataVolumeTemplates {
+		if !searchDV(cdv.Name, newTemplates) {
+			deletedDataVolumes = append(deletedDataVolumes, cdv.Name)
+		}
+	}
+	return deletedDataVolumes
+}
+
+func getDVTIdx(v kubevirtv1.Volume, snapshotVM *snapshotv1.VirtualMachine) int {
+	templateIndex := -1
+	for i, dvt := range snapshotVM.Spec.DataVolumeTemplates {
+		if v.DataVolume.Name == dvt.Name {
+			templateIndex = i
+			break
+		}
+	}
+	return templateIndex
+}
```

Comment:
We can simplify this a bit:
```suggestion
func getDVTIdx(v kubevirtv1.Volume, snapshotVM *snapshotv1.VirtualMachine) int {
        for i, dvt := range snapshotVM.Spec.DataVolumeTemplates {
                if v.DataVolume.Name == dvt.Name {
                        return i
                }
        }       
        return -1
}
```